NetCasbin/CoreEnforcer.cs

```diff
@@ -348,6 +348,18 @@ public void BuildRoleLinks()
             model.BuildRoleLinks(roleManager);
         }
 
+        /// <summary>
+        /// Decides whether a "subject" can access a "object" with the operation
+        /// "action", input parameters are usually: (sub, obj, act).
+        /// </summary>
+        /// <param name="requestValues">The request needs to be mediated, usually an array of strings, 
+        /// can be class instances if ABAC is used.</param>
+        /// <returns>Whether to allow the request.</returns>
+        public Task<bool> EnforceAsync(params object[] requestValues)
```

Comment:
Only wrap a `Task.FromResult` to make this method async may not make this `EnforceAsync` really async. I think to make a real async method, you need to also add async methods for `model` and `adapter`.

Also because `Enforce` is mostly working in memory, so make `Enforce` async seems not much help. I also didn't found any lock implementations when updating policies, if `Enforce` has to wait for a lock, then making it async will have great meaning.

Reply:
The watcher model and adapter already support async. this change is preparation for #76 . we will use lock in the future.

Reply:
Yes, `Enforce` is mostly working in memory, this is `CPU-bound code`. So we can try using the `Task.Run` method to support async.

Reply:
I tried it, but I think It will write and read tokens and parameters at `IExpressionHandler` concurrently, and it may return a wrong result when the model or policy is changed when enforcing.  We can use `Task.Run` when we resolve those questions.